c++/src/kj/compat/http.c++

```diff
@@ -6199,15 +6199,17 @@ private:
 
       auto pipe = newTwoWayPipe();
 
-      kj::Own<kj::AsyncIoStream> io =
-          kj::heap<DelayedEofIoStream>(kj::mv(pipe.ends[0]),
-              task.attach(kj::addRef(*this), kj::mv(headersCopy)));
+      // kj::Own<kj::AsyncIoStream> io =
+      //     kj::heap<DelayedEofIoStream>(kj::mv(pipe.ends[0]),
+      //         task.attach(kj::addRef(*this), kj::mv(headersCopy)));
 
-      fulfiller->fulfill(HttpClient::ConnectResponse {
+      // TODO: Should we make the above customisable?
```

Comment:
Running into the same thing as in https://github.com/capnproto/capnproto/pull/1567/files#r1013083036 here.

I worked around this by creating my own custom implementation of `HttpService::ConnectResponse` which is essentially a clone of this code but without the `DelayedEofIoStream` (you can see that here in my workerd PR: https://github.com/cloudflare/workerd/pull/162/files#diff-4e2699e03d914d2336177c99e165ed8e7f912f9bb61af226be284e62c3324f6eR12).

So since I would have to do the same thing here, I'm wondering whether I can instead add a boolean flag to disable the `DelayedEofIoStream`. That way saving code duplication.

Thinking about it some more though... I'm probably better off moving the `TCPConnectResponseImpl` here.

Reply:
This is what I ended up doing. @kentonv any problems with doing it this way?